src/Utilities/PrettyType.hpp

```diff
@@ -739,4 +739,30 @@ std::string name(const T& /*unused*/) {
   return name<T>();
 }
 /// @}
+
+/*!
+ * \ingroup PrettyTypeGroup
+ * \brief Return a comma separated list of the `pretty_type::name` of every type
+ * in a `tmpl::list`.
+ *
+ * \note The `tmpl::list` must be flattened.
+ */
+template <typename List>
+std::string list_of_names() {
+  static_assert(tt::is_a_v<tmpl::list, List>);
+  std::stringstream ss{};
+  tmpl::for_each<List>([&ss](auto v) {
+    using type = tmpl::type_from<decltype(v)>;
+    static_assert(not tt::is_a_v<tmpl::list, type>,
+                  "The tmpl::list provided to pretty_type::list_of_names must "
+                  "be flattened.");
+    if constexpr (tmpl::index_of<List, type>::value != 0) {
```

Comment:
This breaks if the first type appears multiple times.  There are compile-time ways to do this, but I'd just capture a bool.